code/modules/mob/living/carbon/carbon.dm

```diff
@@ -484,55 +486,54 @@ GLOBAL_LIST_INIT(ventcrawl_machinery, list(/obj/machinery/atmospherics/unary/ven
 
 
 	if(!vent_found)
-		for(var/obj/machinery/atmospherics/machine in range(1,src))
+		for(var/obj/machinery/atmospherics/machine in range(1, src))
 			if(is_type_in_list(machine, GLOB.ventcrawl_machinery) && machine.can_crawl_through())
 				vent_found = machine
 				break
 
-	if(vent_found)
-		if(vent_found.parent && (vent_found.parent.members.len || vent_found.parent.other_atmosmch))
-			visible_message("<span class='notice'>[src] begins climbing into the ventilation system...</span>", \
-							"<span class='notice'>You begin climbing into the ventilation system...</span>")
+	if(!vent_found)
+		to_chat(src, "<span class='warning'>This ventilation duct is not connected to anything!</span>")
+		return
 
-			if(!do_after(src, 45, target = src))
-				return
+	if(vent_found.parent && (length(vent_found.parent.members) || vent_found.parent.other_atmosmch))
+		visible_message("<span class='notice'>[src] begins climbing into the ventilation system...</span>", \
+						"<span class='notice'>You begin climbing into the ventilation system...</span>")
 
-			if(has_buckled_mobs())
-				to_chat(src, "<span class='warning'>You can't vent crawl with other creatures on you!</span>")
-				return
+		if(!do_after(src, 4.5 SECONDS, target = src))
+			return
 
-			if(buckled)
-				to_chat(src, "<span class='warning'>You cannot crawl into a vent while buckled to something!</span>")
-				return
+		if(!vent_found.can_crawl_through())
+			to_chat(src, "<span class='warning'>You can't vent crawl through that!</span>")
+			return
 
-			if(!client)
-				return
+		if(has_buckled_mobs())
+			to_chat(src, "<span class='warning'>You can't vent crawl with other creatures on you!</span>")
+			return
 
-			if(iscarbon(src) && contents.len && ventcrawlerlocal < 2)//It must have atleast been 1 to get this far
-				for(var/obj/item/I in contents)
-					var/failed = 0
-					if(istype(I, /obj/item/bio_chip))
-						continue
-					if(istype(I, /obj/item/reagent_containers/patch))
-						continue
-					if(I.flags & ABSTRACT)
-						continue
-					else
-						failed++
-
-					if(failed)
-						to_chat(src, "<span class='warning'>You can't crawl around in the ventilation ducts with items!</span>")
-						return
+		if(buckled)
+			to_chat(src, "<span class='warning'>You cannot crawl into a vent while buckled to something!</span>")
+			return
 
-			visible_message("<b>[src] scrambles into the ventilation ducts!</b>", "You climb into the ventilation system.")
-			var/old_loc = loc
-			loc = vent_found
-			Moved(old_loc, get_dir(old_loc, loc), FALSE)
-			add_ventcrawl(vent_found)
+		if(!client)
+			return
 
-	else
-		to_chat(src, "<span class='warning'>This ventilation duct is not connected to anything!</span>")
+		if(iscarbon(src) && length(contents) && ventcrawlerlocal < VENTCRAWLER_ALWAYS) // If we're here you can only ventcrawl while completely nude
```

Comment:
do we really need to do this on basetype?

Reply:
I honestly don't know why this proc, which is on `/mob/living` is even in `carbon.dm`